engines/private/private.cpp

```diff
@@ -0,0 +1,1086 @@
+    surf->fillRect(screenRect, _transparentColor);
+    Graphics::Surface *csurf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    surf->transBlitFrom(*csurf, Common::Point(x,y));
+    csurf->free();
+    delete csurf;
+    _image->destroy();
+
+    if (drawn) {
+        drawMask(surf);
+    }
+
+    return surf;
+}
+
+void PrivateEngine::drawMask(Graphics::ManagedSurface *surf) {
+    _compositeSurface->transBlitFrom(surf->rawSurface(), *_origin, _transparentColor);
+    //drawScreen();
+}
+
+void PrivateEngine::drawScreen() {
+    Graphics::ManagedSurface *surface = _compositeSurface;
+
+    if (_videoDecoder) {
+        const Graphics::Surface *frame = _videoDecoder->decodeNextFrame();
+        //frame->create(_videoDecoder->getWidth(), _videoDecoder->getHeight(), _pixelFormat);
+        //frame->copyFrom(*_videoDecoder->decodeNextFrame());
+        Graphics::Surface *cframe = frame->convertTo(_pixelFormat, _videoDecoder->getPalette());
+        Common::Point center((_screenW - _videoDecoder->getWidth())/2, (_screenH - _videoDecoder->getHeight())/2);
+        surface->transBlitFrom(*cframe, center);
+        //frame->free();
+        cframe->free();
+        //delete frame;
+        delete cframe;
+    }
+
+    Graphics::Surface *screen = g_system->lockScreen();
+    if (_mode == 1) {
+        drawScreenFrame(screen);
+    }
+    Common::Rect window(_origin->x, _origin->y, _screenW - _origin->x, _screenH - _origin->y);
+    screen->copyRectToSurface(*surface, _origin->x, _origin->y, window);
+    g_system->unlockScreen();
+    //if (_image->getPalette() != nullptr)
+    //    g_system->getPaletteManager()->setPalette(_image->getPalette(), _image->getPaletteStartIndex(), _image->getPaletteColorCount());
+    g_system->updateScreen();
+
+}
+
+bool PrivateEngine::getRandomBool(uint p) {
+    uint r = _rnd->getRandomNumber(100);
+    return (r <= p);
+}
+
+Common::String PrivateEngine::getPaperShuffleSound() {
+    uint r = _rnd->getRandomNumber(6);
+    return Common::String::format("%sglsfx0%d.wav", _globalAudioPath.c_str(), kPaperShuffleSound[r]);
+}
+
+Common::String PrivateEngine::getTakeSound() {
+    if (isDemo())
+        return (Common::String(_globalAudioPath + "mvo007.wav"));
+
+    uint r = _rnd->getRandomNumber(4) + 1;
+    return Common::String::format("%stook%d.wav", _globalAudioPath.c_str(), r);
+}
+
+Common::String PrivateEngine::getTakeLeaveSound() {
+    uint r = _rnd->getRandomNumber(1);
+    if (r == 0) {
+        return (Common::String(_globalAudioPath + "mvo001.wav"));
+    } else {
+        return (Common::String(_globalAudioPath + "mvo006.wav"));
+    }
+}
+
+Common::String PrivateEngine::getLeaveSound() {
+    if (isDemo())
+        return (Common::String(_globalAudioPath + "mvo008.wav"));
+
+    uint r = _rnd->getRandomNumber(4) + 1;
+    return Common::String::format("%sleft%d.wav", _globalAudioPath.c_str(), r);
+}
+
+char *PrivateEngine::getRandomPhoneClip(const char *clip, int i, int j) {
```

Comment:
I'd recommend changing it to return `Common::String` and avoid explicit memory management.

Reply:
Fixed.